lambda_functions/delivery_status_processor_lambda/delivery_status_processor_lambda.py

```diff
@@ -0,0 +1,175 @@
+        logger.debug(celery_task_body)
+
+        push_to_sqs(celery_task_body, DELIVERY_STATUS_RESULT_TASK_QUEUE, True)
+        
+        logger.info("Processing of request complete")
+
+        return {
+            'statusCode': 200,
+        }
+        
+    except Exception as e:
+        # Place request on dead letter queue so that it can be analyzed 
+        #   for potential processing at a later time
+        logger.error(f'Unknown Failure: {e}')
+        push_to_sqs(event, DELIVERY_STATUS_RESULT_TASK_QUEUE_DEAD_LETTER, False)
+    
+def valid_event(event: dict) -> bool:
+    """
+    Ensure that event data is from the ALB
+    """
+
+    if event is None or "requestContext" not in event or "body" not in event or "elb" not in event["requestContext"] or "headers" not in event or "user-agent" not in event["headers"]:
+        return False
+
+    return True
+
+def event_to_celery_body_mapping(event:dict) -> str:
+    """
+    Determines which SQS queue to send the message to based on the message type
+    """
+    if 'TwilioProxy' in event["headers"]["user-agent"]:
+        return { "body": event["body"], "provider": "twilio" }
+    else:
+        return None
+
+def celery_body_to_celery_task(task_message: dict) -> dict:
+    task = {
+        "task": CELERY_TASK,
+        "id": str(uuid.uuid4()),
+        "args": [
+            {
+                "Message": task_message
+            }
+        ],
+        "kwargs": {},
+        "retries": 0,
+        "eta": None,
+        "expires": None,
+        "utc": True,
+        "callbacks": None,
+        "errbacks": None,
+        "timelimit": [
+            None,
+            None
+        ],
+        "taskset": None,
+        "chord": None
+    }
+    envelope = {
+        "body": base64.b64encode(bytes(json.dumps(task), 'utf-8')).decode("utf-8"),
+        "content-encoding": "utf-8",
+        "content-type": "application/json",
+        "headers": {},
+        "properties": {
+            "reply_to": str(uuid.uuid4()),
+            "correlation_id": str(uuid.uuid4()),
+            "delivery_mode": 2,
+            "delivery_info": {
+                "priority": 0,
+                "exchange": "default",
+                "routing_key": ROUTING_KEY
+            },
+            "body_encoding": "base64",
+            "delivery_tag": str(uuid.uuid4())
+        }
+    }
+
+    return envelope
+
+def push_to_sqs(push_data: dict, queue_url: str, encode: bool) -> None:
+    """
+    Pushes an inbound sms or entire event to SQS. Sends to RETRY or DEAD LETTER queue dependent
+    on is_retry variable. 
+    """
+
+    logger.warning("Pushing to the %s queue . . .", queue_url)
+    logger.debug("SQS push data of type %s: %s", type(push_data), push_data)
+
+    try:
+        if (encode):
+            queue_msg = base64.b64encode(bytes(json.dumps(push_data), 'utf-8')).decode("utf-8")
+        else:
+            queue_msg = json.dumps(push_data)
+    except TypeError as e:
```

Comment:
Are we certain we will only receive `TypeError` here? We have a lot going on in this try block.